src/lib/y2s390/format_process.rb

```diff
@@ -0,0 +1,129 @@
+require "yast"
+require "yast2/execute"
+
+module Y2S390
+  class FormatStatus
+    attr_accessor :progress, :cylinders, :dasd
+    # Constructor
+    #
+    # @param dasd [Dasd]
+    # @param cylinders [Ingeger]
+    # @param format_size [Integer]
+    def initialize(dasd, cylinders, format_size = 10)
+      @dasd = dasd
+      @cylinders = cylinders
+      @progress = 0
+      @size = format_size
+    end
+
+    def update_progress
+      @progress += @size
+    end
+
+    def done?
+      @cylinders <= @progress
+    end
+
+    def step!
+      update_progress
+    end
+  end
+
+  # This class is responsible of formatting a set of DASD volumes maintaining also the status of the
+  # progress.
+  class FormatProcess
+    include Yast::Logger
+    attr_accessor :id, :dasds, :summary, :updated
+
+    SIZE = 10
+
+    # Constructor
+    #
+    # @param dasds [Integer]
+    def initialize(dasds)
+      @dasds = dasds
+      @summary = {}
+      @updated = {}
+    end
+
+    def disks_params
+      dasds.map { |d| "-f /dev/#{d.device_name}" }.join(" ")
+    end
+
+    # Convenience method to start with the formatting process
+    def start
+      cmd = "/sbin/dasdfmt -Y -P #{dasds.size} -b 4096 -y -r #{SIZE} -m #{SIZE} #{disks_params}"
```

Comment:
NP: As in other places, lets put the binary path as a private constant at the beginning of the class (for visibility)